enterprise/frontend/src/embedding/data-picker/DataSelector/DataSelector.jsx

```diff
@@ -366,16 +369,29 @@ export class UnconnectedDataSelector extends Component {
     } else if (
       // Schema id is explicitly set when going through the New > Question/Model flow,
       // whereas we have to obtain it from the state when opening a saved question.
-      (this.state.selectedSchemaId || this.state.selectedSchema?.id) &&
-      steps.includes(TABLE_STEP)
+      this.state.selectedSchemaId ||
+      this.state.selectedSchema?.id
     ) {
       await this.switchToStep(TABLE_STEP);
-    } else if (this.state.selectedDatabaseId && steps.includes(SCHEMA_STEP)) {
-      await this.switchToStep(SCHEMA_STEP);
-    } else if (steps[0] === DATA_BUCKET_STEP && !this.hasUsableModels()) {
-      await this.switchToStep(steps[1]);
+    } else if (this.isJoinStep()) {
```

Comment:
the previous condition was `this.state.selectedDatabase && steps.includes(SCHEMA_STEP)`. The `steps.includes(SCHEMA_STEP)` is always true, and `this.state.selectedDatabase` only has value when it's the join step, so I change the condition to make it more obvious.